### ISSUE TITLE
partitions not properly wiped

### ISSUE THREAD

**Issue by ydirson** [xapi-project/sm#624]
In reports like https://github.com/xcp-ng/xcp/issues/390 and https://xcp-ng.org/forum/topic/7465/upgrade-to-8-3-beta-1-fails we see users getting hit by a previous ZFS partition not being properly wiped.

`util.zeroOut` uses `dd` to clear explicit block ranges in a given block dev (which cannot account for all kinds of signatures in the wild), and tools like `vgcreate` are used in a mode that would ask the user what to do.  Would there be any problem in calling `wipefs -a` on the target block device?

I can see it could be a bad idea to use `--force` on `vgcreate`, as it is quite an "override all checks" flag and there don't seem to be a flag to just ignore previous contents, although it could avoid any upcoming issues that would not be covered by the current `wipefs` version.

**Comment by MarkSymsCtx** [comment]
I think a call to `wipefs -a` sounds entirely reasonable to resolve this issue.

**Comment by ydirson** [comment]
It would seem useful for all SR types, so the `"sr_create"` handling in `SRCommand.run` would be the place, right?

**Comment by MarkSymsCtx** [comment]
> ~It would seem useful for all SR types, so the `"sr_create"` handling in `SRCommand.run` would be the place, right?~
> 
> ... except for all those to which it would not apply

Indeed, I suspect you'd get errors on File based SRs. It could probably go in the LVHDSR class though.

**Comment by ydirson** [comment]
I'm going with `createVG`, so `EXTSR` can benefit from it too
